content/60_gitops_for_socks_shop/30_declare_sock_shop_resource.md

```diff
@@ -0,0 +1,97 @@
++++
+title = "Declare an ArgoCD Helm Chart Object"
+chapter = false
+weight = 30
++++
+
+You'll now declare a [helm chart](https://kubernetes.io/blog/2016/10/helm-charts-making-it-simple-to-package-and-deploy-apps-on-kubernetes/),
+which deploys all of the components that is required to deploy ArgoCD.
+
+Next, let's register an Argo Application in the Kuberletes  chart:
+
+```typescript
+const sockshop = new k8s.apiextensions.CustomResource(
+    "sock-shop",
+    {
+        apiVersion: "argoproj.io/v1alpha1",
+        kind: "Application",
+        metadata: {
+            namespace: "default", // the ns where argocd is deployed
+            name: "sock-shop", // name of app in ArgoCd
+        },
+        spec: {
+            destination: {
+                namespace: ns.metadata.name,
+                server: "https://kubernetes.default.svc",
+            },
+            project: "default",
+            source: {
+                path: "sock-shop",
+                repoURL: "https://github.com/argoproj/argocd-example-apps",
+                targetRevision: "HEAD",
+            },
+            syncPolicy: {
+                automated: {}
+            }
+        }
+    },
+    {provider: provider, parent: ns}
+);
+
+```
+
+{{% notice info %}}
+The `index.ts` file should now have the following contents:
+{{% /notice %}}
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as k8s from "@pulumi/kubernetes";
+
+const pulumiConfig = new pulumi.Config();
+
+// Existing Pulumi stack reference in the format:
+// <organization>/<project>/<stack> e.g. "myUser/myProject/dev"
+const clusterStackRef = new pulumi.StackReference(pulumiConfig.require("clusterStackRef"));
+const argoStackRef = new pulumi.StackReference(pulumiConfig.require("argoCDStackRef"))
+
+// Get the kubeconfig from the cluster stack output.
+const kubeconfig = clusterStackRef.getOutput("kubeconfig");
+const argoCDNamespace = argoStackRef.getOutput("argoNamespace");
+
+// Create the k8s provider with the kubeconfig.
+const provider = new k8s.Provider("k8sProvider", {kubeconfig});
+
+// Declare a Namespace in which to deploy argocd
+let name = "argocd"
+const ns = new k8s.core.v1.Namespace("argocd-ns", {
+    metadata: { name: name },
+}, { provider });
+
+const sockshop = new k8s.apiextensions.CustomResource(
+    "sock-shop",
+    {
+        apiVersion: "argoproj.io/v1alpha1",
+        kind: "Application",
+        metadata: {
+            namespace: argoCDNamespace.apply(x => x.toString()), // the ns where argocd is deployed
```

Comment:
Not sure if the namespaces all line up here. It's `default` in the `CustomResource` above, then you create a `argocd` namespace, and then this is reading from a config.

Reply:
Fixed! Great spot